src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -202,6 +216,25 @@ ss::future<ss::stop_iteration> leader_balancer::balance() {
         return now >= g.second.expires;
     });
 
+    absl::erase_if(
+      _in_flight_changes, [this, now = clock_type::now()](const auto& c) {
+          const auto& [group, change] = c;
+
+          if (now >= change.expires) {
+              _probe.leader_transfer_timeout();
+              vlog(
+                clusterlog.info,
+                "Leadership movement for group {} from {} to {} timed out",
```

Comment:
Let's re-word this a bit while we're changing the code: at this point we've already seen a successful transfer, and the timeout is of the propagation of leadership updates, rather than the transfer itself.

Reply:
Reworded it. Let me know if the new wording works for you.